packages/block-editor/src/store/private-actions.js

```diff
@@ -158,23 +153,22 @@ export const privateRemoveBlocks =
 		// confirmation that they intended to remove such block(s). However,
 		// the editor instance is responsible for presenting those confirmation
 		// prompts to the user. Any instance opting into removal prompts must
-		// register using `toggleRemovalPromptSupport()`.
+		// register using `setBlockRemovalRules()`.
 		//
 		// @see https://github.com/WordPress/gutenberg/pull/51145
-		if ( ! forceRemove && select.isRemovalPromptSupported() ) {
+		let rules;
+		if ( ! forceRemove && ( rules = select.getBlockRemovalRules() ) ) {
```

Comment:
Why assign inside the `if`?

Reply:
I didn't want to call the selector before examining the forceRemove flag, and I thought this was nicer than nesting two ifs. Do you think I should avoid this? Does it not read well?

Reply:
No big deal of course, though. If you want just merge as is :) .

Reply:
You're right, that suits Gutenberg better. I'm bringing old C idioms for no good reason, maybe. :) Thanks, I'll push and merge. 🚀